src/pymmcore_widgets/mda/_save_widget.py

```diff
@@ -0,0 +1,182 @@
+        writer = self._get_writer_from_extension(extension)
+
+        # if it's a valid writer, just update the combo
+        if extension and writer in WRITERS:
+            self._update_combo_text(writer)
+
+        # if the extension is not valid, clear the extension and get it form the combo
+        elif name := self.save_name.text():
+            self.save_name.setText(name + WRITERS[self._writer_combo.currentText()][0])
+
+    def _get_extension_from_name(self, name: str) -> str:
+        """Get the extension of the selected writer."""
+        for ext in EXTENSIONS:
+            if name.endswith(ext):
+                return ext
+        return ""
+
+    def _get_writer_from_extension(self, extension: str) -> str:
+        """Get the writer from the extension."""
+        for writer, ext in WRITERS.items():
+            for e in ext:
+                if extension == e:
+                    return writer
+        raise ValueError(f"Extension {extension} not found in {WRITERS}")
+
+    def _update_combo_text(self, writer: str) -> None:
+        """Update the writer combo and trigger the text changed signal."""
+        # blocking and then manually calling _on_writer_combo_text_changed because
+        # if the text of the combo does not change, the currentTextChanged signal is not
+        # emitted and we need it to add the extension to the name
+        with signals_blocked(self._writer_combo):
+            self._writer_combo.setCurrentText(writer)
+        self._on_writer_combo_text_changed(writer)
+
+    def _on_writer_combo_text_changed(self, writer: str) -> None:
+        """Update the save name extension when the writer is changed."""
+        self.name_label.setText(SUBFOLDER if writer == "tiff-sequence" else FILE_NAME)
+        extension = self._get_extension_from_name(self.save_name.text())
+        if name := self._remove_extension(self.save_name.text()):
+            # get index of the extension in the list or get the first one
+            # this is useful in the case of the ome-tiff writer that has two extensions
+            # .ome.tiff and .ome.tif
+            idx = next(
+                (i for i, ext in enumerate(WRITERS[writer]) if extension == ext), 0
+            )
+            self.save_name.setText(name + WRITERS[writer][idx])
+
+    def _remove_extension(self, name: str) -> str:
+        """Remove the extension from the name if it's there."""
+        return next((name.replace(ext, "") for ext in EXTENSIONS if ext in name), name)
+
+    def value(self) -> str | None:
+        """Return current state of the save widget."""
+        if (
+            not self.isChecked()
+            or not self.save_dir.text()
+            or not self.save_name.text()
+        ):
+            return None
+
+        # returning a str because the MDASequence yaml does not support Path yet
+        # maybe we can add support for Path in the future
+        return str(Path(self.save_dir.text()) / self.save_name.text())
```

Comment:
i'm not sure about this change, reducing the value of the of the widget down to a single string.  The dict before was more explicit, and encoded the state of the widget better.   Can you comment on why you reduced it down to a single string?

as for the conversion from string to yaml.  that should only happen when placing the object into a useq.MDASequence.metadata dict.  In other words, don't do it here in this widget just in case some one elsewhere wants to put the output of `widget.value()` into a useq metadata and then later serialize it.  make sense?  that's thinking to specifically about what one might do with the return value from this method

Reply:
I thought it was simpler to just pass a string instead of a dict with "save_dir" and  "save_name". Do you think it's better to use the dict? what if we use the `tiff-sequence`, should we change /add a key "subfolder"?

Reply:
if we use a dict, there's no question about what was in the top field, and what was in the bottom (path) field.  If we use a string, it's up to the receiver to split them.  And in most cases, they'll probably do it correctly, (but still inasmuch as they have to split them anyway, we might as well just give them split).  You could also, for example, include the "requested format" from the combo box in that dict, but if you use a string, the receiver again needs to make a guess based on the extension of that one string.

Reply:
> what if we use the tiff-sequence, should we change /add a key "subfolder"?

no, don't change the keys in the dict based on their values.  Just keep the keys as whatever they are in the first place (the fact that we change the Label in the GUI is just a convenience for the GUI user, it doesn't need to have any bearing on the fact that this widget is still just a two field widget)